tools/cli/commands/create.py

```diff
@@ -358,80 +383,108 @@ def flags(parser):
       gcloud_compute: Function that can be used for invoking `gcloud compute`
+      network_name: The name of the network
     Raises:
       subprocess.CalledProcessError: If the `gcloud` command fails
     """
     if utils.print_info_messages(args):
-        print('Creating the network {0}'.format(_DATALAB_NETWORK))
+        print('Creating the network {0}'.format(network_name))
     create_cmd = [
-        'networks', 'create', _DATALAB_NETWORK,
+        'networks', 'create', network_name,
         '--description', _DATALAB_NETWORK_DESCRIPTION]
     utils.call_gcloud_quietly(args, gcloud_compute, create_cmd)
     return
 
 
-def ensure_network_exists(args, gcloud_compute):
-    """Create the `datalab-network` network if it does not already exist.
+def ensure_network_exists(args, gcloud_compute, network_name):
+    """Create the specified network if it does not already exist.
 
     Args:
       args: The Namespace returned by argparse
       gcloud_compute: Function that can be used for invoking `gcloud compute`
+      network_name: The name of the network
     Raises:
       subprocess.CalledProcessError: If the `gcloud` command fails
     """
-    get_cmd = ['networks', 'describe', '--format', 'value(name)',
-               _DATALAB_NETWORK]
+    get_cmd = ['networks', 'describe', '--format', 'value(name)', network_name]
     try:
         utils.call_gcloud_quietly(
             args, gcloud_compute, get_cmd, report_errors=False)
     except subprocess.CalledProcessError:
-        create_network(args, gcloud_compute)
+        create_network(args, gcloud_compute, network_name)
     return
 
 
-def create_firewall_rule(args, gcloud_compute):
-    """Create the `datalab-network-allow-ssh` firewall rule.
+def create_firewall_rule(args, gcloud_compute, network_name, rule_name):
+    """Create the `{rule_name}` firewall rule.
 
     Args:
       args: The Namespace returned by argparse
       gcloud_compute: Function that can be used for invoking `gcloud compute`
+      network_name: The name of the network on which to allow SSH access
+      rule_name: The name of the firewall rule
     Raises:
       subprocess.CalledProcessError: If the `gcloud` command fails
     """
     if utils.print_info_messages(args):
-        print('Creating the firewall rule {0}'.format(_DATALAB_FIREWALL_RULE))
+        print('Creating the firewall rule {0}'.format(rule_name))
     create_cmd = [
-        'firewall-rules', 'create', _DATALAB_FIREWALL_RULE,
+        'firewall-rules', 'create', rule_name,
         '--allow', 'tcp:22',
-        '--network', _DATALAB_NETWORK,
+        '--network', network_name,
         '--description', _DATALAB_FIREWALL_RULE_DESCRIPTION]
     utils.call_gcloud_quietly(args, gcloud_compute, create_cmd)
     return
 
 
-def ensure_firewall_rule_exists(args, gcloud_compute):
-    """Create the `datalab-network-allow-ssh` firewall rule if it necessary.
+def has_non_standard_firewall_rules(args, gcloud_compute, network_name):
```

Comment:
It is not obvious to me what "non-standard" means here. Maybe add a comment stating that this returns true if there are any firewall rules other than the one we expect.

Reply:
Changed to "unexpected". Is that more clear?

Reply:
Yes, that's better. Thanks.